lib/src/components/button/Button.tsx

```diff
@@ -219,60 +227,59 @@ const WithLoader = ({
   </>
 )
 
-type ButtonProps = Override<
-  React.ComponentProps<typeof StyledButton>,
+type ButtonProps<
+  H extends string | undefined,
+  C extends React.ElementType
```

Comment:
Optional parameter should be at the right.

```suggestion
  C extends React.ElementType,
  H extends string | undefined,
```

Otherwise, we might see types like - `ButtonProps<undefined, ...>`

Reply:
These should always be taken from the `href` and `as` props and never manually defined.